uqcsbot/scripts/bgg.py

```diff
@@ -0,0 +1,149 @@
+        # sets the user ratings
+        elif i.tag == "statistics":
+            for j in i[0]:
+                if j.tag == "average":
+                    parameters["score"] = j.attrib["value"]
+                if j.tag == "usersrated":
+                    parameters["users"] = j.attrib["value"]
+                if j.tag == "ranks":
+                    for k in j:
+                        if (k.attrib.get("name") == "boardgame"
+                                and k.attrib.get("value").isnumeric()):
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            parameters["rank"] = f"{n:d}{o:s}"
+                        elif k.attrib.get("value").isnumeric():
+                            n = int(k.attrib.get("value"))
+                            o = "tsnrhtdd"[(n/10 % 10 != 1) * (n % 10 < 4) * n % 10::4]
+                            s = " ".join(k.attrib["friendlyname"].split(" ")[:-1])
+                            parameters["subranks"][s] = f"{n:d}{o:s}"
+        # sets the discription
+        elif i.tag == "description":
+            parameters["description"] = i.text
+        # sets the minimum playing time
+        elif i.tag == "minplaytime":
+            parameters["min_time"] = i.attrib["value"]
+        elif i.tag == "maxplaytime":
+            parameters["max_time"] = i.attrib["value"]
+
+    return parameters
+
+
+def format_board_game_parameters(parameters: dict) -> str:
+    message = (f"*{parameters.get('name', ':question:'):s}*\n"
+               f"A board game for {parameters.get('min_players', ':question:'):s} to"
+               + f" {parameters.get('max_players', ':question:'):s} players, with a"
+               + f" playing time of {parameters.get('min_time', ':question:'):s} minutes"
+               + ("" if parameters.get('min_time') == parameters.get('max_time')
+                  else f" to {parameters.get('max_time', ':question:'):s} minutes") + ".\n"
+               f"Rated {parameters.get('score', ':question:'):s}/10 by"
+               + f" {parameters.get('users', ':question:'):s} users.\n"
+               f"Ranked {parameters.get('rank', ':question:'):s} overall on _Board Game Geek_.\n"
+               + "".join(f"• Ranked {value:s} in the {key:s} category.\n"
+                         for key, value in parameters.get("subranks", {}).items()) +
+               f"Categories: {', '.join(parameters.get('categories', set())):s}\n"
+               f"Mechanics: {', '.join(parameters.get('mechanics', set())):s}\n"
+               f"https://boardgamegeek.com/boardgame/{parameters.get('identity', ':question:'):s}\n"
+               f">>>{parameters.get('description', ':question:'):s}")
+    message = unescape(message)
+    while "\n\n" in message:
+        message = message.replace("\n\n", "\n")
+    if len(message) > 4000:
+        message = message[:3999] + "…"
```

Comment:
This is probably an edge case but maybe just character limit the description, to avoid truncating arbitrarily in the middle of the other info. Also, 4000 characters is a bit long...

Reply:
The maximum length of a Slack message is 4'000 characters, and if that means the message will be cut off in the middle of some other information, so be it.  
Also, it will only cut off information other than the description if the game is in, like, 80%+ of the available genres, categories and mechanics, which is exceedingly unlikely.